pkg/tinydtls/include/sock_dtls_types.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief   tinydtls-specific types and functions definitions
+ *
+ * @author  Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ * @author  Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+
+#ifndef SOCK_DTLS_TYPES_H
+#define SOCK_DTLS_TYPES_H
+
+#include "dtls.h"
+#include "net/sock/udp.h"
+#include "net/credman.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef SOCK_DTLS_MBOX_SIZE
+#define SOCK_DTLS_MBOX_SIZE     (4)         /**< Size of DTLS sock mailbox */
+#endif
+
+/**
+ * @brief Information about DTLS sock
+ */
+struct sock_dtls {
+    dtls_context_t *dtls_ctx;               /**< TinyDTLS context for sock */
+    sock_udp_t *udp_sock;                   /**< Underlying UDP sock to use */
```

Comment:
What was the reason for not doing
```suggestion
    sock_udp_t udp_sock;                    /**< Underlying UDP sock to use */
    dtls_context_t *dtls_ctx;               /**< TinyDTLS context for sock */
```
instead?

That way application code using UDP sockets could transparently handle DTLS sockets (after a custom setup procedure).
`sock_udp_t` would only need to be extended by a flag to indicate there is a DTLS context attached to it.

Reply:
We might want to support other transports in the future. With that, it would be easier and better understandable to create the transport sock in the application first and then the DTLS sock. As we do not want to copy that transport sock, a pointer is the best way to go here.